### DISCUSSION TITLE
Raw IQ sample demodulation

### DISCUSSION THREAD

**Discussion by pwillen** [szpajder/dumpvdl2#49] [Q&A]
Hello, 
I am generating my own raw iq files. I am struggling to verify the correct format for dumpvdl2 processing. 

How is the example vdl2_model_16b_1050kHz.wav structured? Passing it through scipy.wavefile.read yields a 2darray. Initially I thought the arrays were the real and imaginary portions of the signal. I think that is wrong...Are these two different channels?

Is the example file already AM demodulated as well?

**Comment by szpajder** [comment]
vdl2_model_16b_1050kHz.wav contains I/Q data. It's a stereo wave file,
signed 16-bit samples, left channel = I, right channel = Q. The WAV header
is there just for convenience, so that you can view the file in Audacity or
other waveform editor. It is not meant to be played, because it's not an
audio recording. dumpvdl2 does not need the WAV header, raw I/Q data file
without any header will do fine. More info:
https://github.com/szpajder/dumpvdl2#processing-recorded-iq-data-from-file

**Reply by pwillen** [reply]
Thank you for the quick response. I was struggling to get any decodes from the samples that I have and wanted to make sure formatting as well as the cli commands were correct first and foremost. I was able to get some success eventually! Seems like any issues I was having was due to the quality of the sample file.
